File: src/api/domain/category.go
package domain

import "go.mongodb.org/mongo-driver/bson/primitive"

type Category struct {
	Id     primitive.ObjectID `json:"id" bson:"_id" required:"false"`
	Name   string             `json:"name" bson:"name" required:"false"`
	Hidden bool               `json:"hidden" bson:"hidden" required:"false"`
	Icon   string             `json:"iconClass" bson:"icon" required:"false"`
}

File: src/ui/src/app/domain/category.ts
export class Category {
  id: string;
  name: string;
  hidden: boolean;
  iconClass: string; //FontAwesome 5 icons available


  constructor(id: string, name: string, hidden: boolean, iconClass: string) {
    this.id = id;
    this.name = name;
    this.hidden = hidden;
    this.iconClass = iconClass;
  }
}
